src/tools/rest-backend/root_application.cpp

```diff
@@ -0,0 +1,356 @@
+#include <boost/algorithm/string/replace.hpp>
+
+#include "root_application.hpp"
+
+
+namespace kdbrest
+{
+
+/**
+     * @brief Constructor for main service class
+     *
+     * @param srv Takes a CppCms service
+     */
+RootApp::RootApp (cppcms::service & srv) : cppcms::application (srv)
+{
+
+	attach (new AuthenticationApp (srv), "auth", "/auth{1}", // mapping
+		"/auth(/(.*))?", 1				 // dispatching
+		);
+
+	attach (new DatabaseApp (srv), "database", "/database{1}", // mapping
+		"/database(/(.*))?", 1				   // dispatching
+		);
+
+	dispatcher ().assign ("/gettest/((([a-zA-Z0-9\\-\\.]+)/?)+)", &RootApp::getTest, this, 1);
+	mapper ().assign ("gettest", "/gettest/{1}");
+
+	dispatcher ().assign ("", &RootApp::welcome, this);
+	mapper ().assign ("");
+
+	mapper ().root ("");
+}
+
+
+void RootApp::welcome ()
+{
+	if (request ().request_method () == "GET")
+	{
+            response().out() << "<h1>Welcome to the Elektra REST service!</h1>\n"
```

Comment:
No, I only added this part for the prototype so that it is clear which functions are available. I already planned so use a (maybe also better looking) template later. Was a quick & dirty solution in the evening.